### PR TITLE
Bug fix for #154

### PR DESCRIPTION
**Pull request by blueset**
* Update school selector filter in "shared/components/sign" to follow the current year instead of hardcoded year numbers.
* Optimised list view of Paths, Levels, and Problems for small view 
 * removed level language tag text
 * ~~added smaller icon of language in front~~
 * Added icon for "Run" button in small view
* UI optimisation: 
 * changed icon of New Path from `add-circle-outline` to `add`
 * ~~changed nav-bar button from `md-raised md-primary` to "flat" button~~
 * Added style for school without second student in `classmentors/components/ace`
 * Fixed phrasing in alert text in `setCodeCombatUserIdCtrlInitialData`
 * Changed buttons in creating new event in ClassMentors from `raised-primary`and `raised` to `primary` and "flat" 
 * Changed all `add-circle-outline` icon to `add` icon
 * Fixed alignments of names in Event Participants Challenges / Ranking View.
 * Changed leaving group icon from `highlight-remove` to `clear`
 * Trying to fix icon overlap issue by changing `row-height` to 3:4 ratio. ( issue #54 )
 
**Since last PR**
* UI optimisation: 
 * Added `.responsive-button` for buttons that hides their caption in small devices
 * Added icons `chevron-right`, `check`, `clear`, `add`
 * Replaced the language icon back only in Paths list (no need of language icon in Levels list and Problems list as they are all in the same language)
 * Replaced buttons to `.md-raised.md-primary` for better identification
 * Fixed bug in school selection that appeared in commit/ ced9885b57f21e600f063328db0c20a8252ad259 
 * Fixed bug in Paths/Levels/Problems list appeared in commit/ 7f3c890ccc325664327e37a3bbc1132009db45bc (found by @dinoboff in issue #154 )
 * Fixed bug in display of "no second student“ in commit/ 5f91ef6c27886c09d656fedc147229b51734b318
 * Further modified `classMentors/profile` page ( issue #54 )
  * Changed the layout of the page to be `layout-wrap`
  * Added responsive feature to the main `flex` div and the school crest `flex` div.
 * Added all missing icons (found by @dinoboff in issue #154 )
* Update test to match the changes

### CODE REVIEW

File: src/shared/components/icons/svgdefs.svg [hunk 1]
```diff
@@ -54,5 +54,17 @@
...
+        </g>
+        <g id="check" viewBox="0 0 1024 1024">
+            <path class="path1" d="M384 690l452-452 60 60-512 512-238-238 60-60z"></path>
+        </g>
+        <g id="clear" viewBox="0 0 1024 1024">
+            <path class="path1" d="M810 274l-238 238 238 238-60 60-238-238-238 238-60-60 238-238-238-238 60-60 238 238 238-238z"></path>
+        </g>
+        <g id="add" viewBox="0 0 1024 1024">
+            <path class="path1" d="M899.725 566.645h-333.077v333.077h-109.292v-333.077h-333.077v-109.292h333.077v-333.077h109.292v333.077h333.077v109.292z"></path>
+        </g>
```
Comment (dinoboff) [inline src/shared/components/icons/svgdefs.svg]: There's already an icon with the clear id.

They seems identical: http://codepen.io/dinoboff/pen/GJaGmv

File: src/shared/components/sign/sign-view.html [hunk 2]
```diff
@@ -50,7 +50,7 @@
     	
         <p ng-if="!ctrl.loaded" flex>Loading schools...</p>
 
-		<md-input-container flex layout="column" ng-if="ctrl.loaded &amp;&amp; currentUser.yearOfBirth >= 1996 &amp;&amp; currentUser.yearOfBirth <= 2004">
+		<md-input-container flex layout="column" ng-if="currentYear - currentUser.yearOfBirth <= 20 &amp;&amp; currentYear - currentUser.yearOfBirth  >= 10">
```
Comment (dinoboff) [inline src/shared/components/sign/sign-view.html]: `ng-if` tests should includes `ctrl.loaded`.

Comment (dinoboff) [inline src/shared/components/sign/sign-view.html]: @blueset @SingaporeClouds 

It's now requiring people born in years between 1995 and 2005 to pick a school. Is it ok?

**Comment by dinoboff** [comment]
Thanks, looking good except for the two small issues.

File: src/shared/components/sign/sign-view.html (hunk 2 above)
Comment (blueset) [inline src/shared/components/sign/sign-view.html]: I think this should be okay. As what I observe from my school, in Junior College Year 2, the year of birth ranges from 1995 to 1998. Adding 6 to 1998 will be 2004.

**Comment by blueset** [comment]
@dinoboff I have fixed them in 7d8bf6bc17cf74992bbd5434fa5adb91e4bf92b2 , thanks for pointing out.

**Comment by dinoboff** [comment]
Looks good to me.

Thanks blueset.

**Pull request merged**